crates/ruff_macros/src/config.rs

```diff
@@ -212,39 +233,109 @@ struct FieldAttributes {
     example: String,
 }
 
-impl Parse for FieldAttributes {
-    fn parse(input: ParseStream) -> syn::Result<Self> {
-        let default = _parse_key_value(input, "default")?;
```

Comment:
The old logic relied on the specific order. The new logic allows arbitrary ordering and is in line with that serde and other crates use for parsing attribute fields.